cmd/kubectl/kubectlcobra/commands.go

```diff
@@ -93,28 +95,20 @@ func NewCmdApply(baseName string, f util.Factory, ioStreams genericclioptions.IO
 		Run: func(cmd *cobra.Command, args []string) {
 			if len(args) > 0 {
 				// check is kustomize, if so update
-				o.DeleteFlags.FileNameFlags.Kustomize = &args[0]
+				applier.applyOptions.DeleteFlags.FileNameFlags.Kustomize = &args[0]
 			}
 
-			cmdutil.CheckErr(o.Complete(f, cmd))
-			cmdutil.CheckErr(o.Run())
-			infos, _ := o.GetObjects()
-			if so.wait {
-				cmdutil.CheckErr(so.waitForStatus(infos))
-			}
+			cmdutil.CheckErr(applier.Initialize(cmd))
+			ch := applier.run(context.Background())
+
+			printer.Print(ch)
```

Comment:
if this blocks on the ch closing, please comment, if not -- also please comment.